src/tests/Microsoft.Diagnostics.NETCore.Client/PerfMapTests.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.CommonTestRunner;
+using Microsoft.Diagnostics.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Extensions;
+using TestRunner = Microsoft.Diagnostics.CommonTestRunner.TestRunner;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public class PerMapTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public static IEnumerable<object[]> Configurations => TestRunner.Configurations;
+
+        public PerMapTests(ITestOutputHelper outputHelper)
+        {
+            _output = outputHelper;
+        }
+
+        private static bool DoFilesExist(PerfMapType type, int pid)
+        {
+            if (type == PerfMapType.All || type == PerfMapType.PerfMap)
+            {
+                string expectedPerfMapFile = GetPerfMapFileName(pid);
+                string expectedPerfInfoFile = GetPerfInfoFileName(pid);
+
+                if (!File.Exists(expectedPerfMapFile) || !File.Exists(expectedPerfInfoFile))
+                {
+                    return false;
+                }
+            }
+
+            if (type == PerfMapType.All || type == PerfMapType.JitDump)
+            {
+                string expectedJitDumpFile = GetJitDumpFileName(pid);
+                if (!File.Exists(expectedJitDumpFile))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static string GetTmpDir()
+        {
+            string tmpDir = Environment.GetEnvironmentVariable("tmpdir");
```

Comment:
```suggestion
            string tmpDir = Environment.GetEnvironmentVariable("TMPDIR");
```

Since env vars are case sensitive on Linux